packages/grid/data-grid/src/tests/pagination.DataGrid.test.tsx

```diff
@@ -67,6 +68,76 @@ describe('<DataGrid /> - Pagination', () => {
       }, 50);
     });
 
+    it('should trigger onPageChange once if page prop is set', () => {
+      const onPageChange = spy();
+
+      render(
+        <div style={{ width: 300, height: 300 }}>
+          <DataGrid {...baselineProps} page={1} onPageChange={onPageChange} pageSize={1} />
+        </div>,
+      );
+      expect(onPageChange.callCount).to.equal(1);
+    });
+
+    it('should trigger onPageChange when clicking on next page', () => {
+      const onPageChange = spy();
+
+      render(
+        <div style={{ width: 300, height: 300 }}>
+          <DataGrid {...baselineProps} onPageChange={onPageChange} pageSize={1} />
+        </div>,
+      );
+      fireEvent.click(screen.getByRole('button', { name: /next page/i }));
+      expect(onPageChange.callCount).to.equal(1);
+    });
+
+    it('should trigger onPageChange when clicking on next page in Server mode', () => {
+      const onPageChange = spy();
+
+      render(
+        <div style={{ width: 300, height: 300 }}>
+          <DataGrid
+            {...baselineProps}
+            onPageChange={onPageChange}
+            pageSize={1}
+            paginationMode={'server'}
+          />
+        </div>,
+      );
+      // First time on init
+      expect(onPageChange.callCount).to.equal(1);
+      fireEvent.click(screen.getByRole('button', { name: /next page/i }));
+      expect(onPageChange.callCount).to.equal(2);
+      fireEvent.click(screen.getByRole('button', { name: /next page/i }));
+      expect(onPageChange.callCount).to.equal(3);
```

Comment:
The default page is 1, it shouldn't fire for it

```suggestion
      expect(onPageChange.callCount).to.equal(0);
      fireEvent.click(screen.getByRole('button', { name: /next page/i }));
      expect(onPageChange.callCount).to.equal(1);
      fireEvent.click(screen.getByRole('button', { name: /next page/i }));
      expect(onPageChange.callCount).to.equal(2);
```

Reply:
I think it should fire to easily load the content of the first page like any other page

Reply:
This avoids to pass the content of the first page in the rows prop. Page 1 is treated like any other page

Reply:
I agree with @oliviertassinari . The description in the bug that was reported was that the expectation is that the event should be fired initially, only after the page change, which I think the reporter understands as interacting with the pagination (from page 1 to page 2 for example).